pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ProducerBuilder.java

```diff
@@ -172,7 +172,7 @@
      * the client application. Until, the producer gets a successful acknowledgment back from the broker,
      * it will keep in memory (direct memory pool) all the messages in the pending queue.
      *
-     * <p>Default is 1000.
+     * <p>Default is 0, disable the pending messages check.
```

Comment:
We should also deprecate `maxPendingMessagesAcrossPartitions()` (and setting it to 0 by default), because memory limit is a better approach to limit the max amount of memory.

Reply:
Done. The `maxPendingMessagesAcrossPartitions ` and `maxPendingMessages` are deprecated.

Reply:
Oh, but let's not deprecate this one (`maxPendingMessages`), since there will still be legitimate reasons to set a max number of messages.

Reply:
OK, done.